src/orc.cpp

```diff
@@ -342,64 +358,157 @@ odrv_report::odrv_report(std::string_view symbol, const die* list_head)
 
     // Construct a map of unique definitions of the conflicting symbol.
 
-    if (_conflict_map.empty()) {
-        for (const die* next_die = _list_head; next_die; next_die = next_die->_next_die) {
-            std::size_t hash = next_die->_fatal_attribute_hash;
-            if (_conflict_map.count(hash)) continue;
-            conflict_details details;
-            details._die = next_die;
-            details._attributes = fetch_attributes_for_die(*details._die);
-            _conflict_map[hash] = std::move(details);
+    for (const die* next_die = _list_head; next_die; next_die = next_die->_next_die) {
+        const std::size_t hash = next_die->_fatal_attribute_hash;
+        const bool new_conflict = _conflict_map.count(hash) == 0;
+        auto& conflict = _conflict_map[hash];
+        auto attributes = fetch_attributes_for_die(*next_die);
+
+        ++conflict._count;
+
+        if (const auto location = derive_definition_location(attributes)) {
+            std::stringstream ss;
+            ss << object_file_ancestry(next_die->_ofd_index);
+            conflict._locations[*location].emplace_back(std::move(ss).str());
+        }
+
+        if (new_conflict) {
+            // The fatal attribute hash should be the same for all instances
+            // of this `conflict`, so we only need to set its attributes once.
+            conflict._attributes = std::move(attributes);
+            conflict._tag = next_die->_tag;
         }
     }
 
     assert(_conflict_map.size() > 1);
 
-    // Derive the ODRV category.
+    // Derive the ODRV categories.
+    const auto conflict_first = _conflict_map.begin();
+    const auto conflict_last = _conflict_map.end();
+
+    for (auto x = conflict_first; x != conflict_last; ++x) {
```

Comment:
Instead of comparing just two entries in the conflict map, now we compare all of them to each other, collecting all the categories where they conflict along the way.